test/parallel/test-http2-invalidheaderfield.js

```diff
@@ -0,0 +1,70 @@
+'use strict';
+const common = require('../common');
+if (!common.hasCrypto) { common.skip('missing crypto'); }
+
+// Check for:
+// Spaced headers
+// Psuedo headers
+// Capitalized headers
+
+const http2 = require('http2');
+const { throws } = require('assert');
+
+const server = http2.createServer(common.mustCall((req, res) => {
+
+  throws(() => {
+    res.setHeader(':path', '/');
+  }, TypeError);
+
+  throws(() => {
+    res.setHeader('t est', 123);
+  }, TypeError);
+
+  // should not fail
+  res.setHeader(' test', 123);
```

Comment:
Equivalent to 'test'.
https://github.com/nodejs/node/blob/a29d7c7ff7ed68cc7d1ba30b8ef24f64dc59df7c/lib/internal/http2/compat.js#L604

Reply:
Hm, I wonder, the `http` fails with `ERR_INVALID_HTTP_TOKEN` if you try to set the header name of `' test'`.

Reply:
> Each header field consists
   of a name followed by a colon (":") and the field value. *Field names*
   are case-insensitive. The *field value* MAY be preceded by any amount
   of LWS, though a single SP is preferred.

@lundibundi The [RFC2616](https://tools.ietf.org/html/rfc2616#section-4.2) for HTTP/1 states that the field value may be preceded by any amount of LWS, but nothing similar is stated for the field names. Furthermore, this is not a problem in HTTP/2 owing to the [binary protocol](https://http2.github.io/faq/#why-is-http2-binary).

Reply:
That part of the RFC if referring to the format of the HTTP message (i.e. actual text representation of the message), not the specific field name/value. Just down below in that section, we can see
` field-name     = token `
And the token is
`token          = 1*<any CHAR except CTLs or separators>`

Therefore there shouldn't be any leading whitespace in the user-provided value for header name which our http/1 implementation is enforcing correctly (by throwing `ERR_INVALID_HTTP_TOKEN`).

It's fine to do that in a follow-up PR, I just mentioned it so that we won't forget about it.